### ISSUE TITLE
Build consistently fails on first build, and succeeds on second.

### ISSUE THREAD

**Issue by jamescrowley** [dotnet/cli#7839]
## Steps to reproduce

1. Download attached zip. [SampleConcurrencyBuildIssue.zip]
[SampleConcurrencyBuildIssue.zip](https://github.com/dotnet/cli/files/1384725/SampleConcurrencyBuildIssue.zip)
2. Run build.cmd (build fails)
3. Run build.cmd (build passes)
4. Rename folder that you've extracted to, or delete contents of obj/bin
5. Run build.cmd (build fails)
6. Run build.cmd (build passes)

## Notes

* I hit this issue while trying to migrate all class libraries in a large (150+ project) solution to the new project format, while for now preserving our existing web projects in the old project format (as we can't migrate that until switching away from MVC 5)
* the sample consists of a library that has been migrated to the new project format, and a web project in the old project format - all targeting .NET 4.6.2.
* The packages have been restored by paket, but I have removed all trace of paket itself. So regardless of what paket generates, I'm pretty confident this is a build toolchain issue.
* switch the build command to run the solution file instead of the csproj, and the build succeeds (in this repro case). However in my underlying case (a solution with 150 projects), the build fails running against the solution

## Expected  behavior

Build should fail or pass consistently.

## Actual behavior

First build:

	C:\Development\Sample>dotnet build src/FundApps.Rapptr.Web/FundApps.Rapptr.Web.csproj
	Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
	Copyright (C) Microsoft Corporation. All rights reserved.

	  FundApps.Common -> C:\Development\Sample\src\FundApps.Common\bin\Debug\net462\FundApps.Common.dll
	C:\Development\Sample\packages\Microsoft.Net.Compilers\tools\Microsoft.CSharp.Core.targets(84,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Development\Sample\packages\Microsoft.Net.Compilers\build\..\tools\Microsoft.Build.Tasks.CodeAnalysis.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Development\Sample\src\FundApps.Rapptr.Web\FundApps.Rapptr.Web.csproj]

	Build FAILED.

	C:\Development\Sample\packages\Microsoft.Net.Compilers\tools\Microsoft.CSharp.Core.targets(84,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Development\Sample\packages\Microsoft.Net.Compilers\build\..\tools\Microsoft.Build.Tasks.CodeAnalysis.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\Development\Sample\src\FundApps.Rapptr.Web\FundApps.Rapptr.Web.csproj]
	    0 Warning(s)
	    1 Error(s)

	Time Elapsed 00:00:05.38

Second build:

	C:\Development\Sample>build

	C:\Development\Sample>dotnet build src/FundApps.Rapptr.Web/FundApps.Rapptr.Web.csproj
	Microsoft (R) Build Engine version 15.4.8.50001 for .NET Core
	Copyright (C) Microsoft Corporation. All rights reserved.

	  FundApps.Common -> C:\Development\Sample\src\FundApps.Common\bin\Debug\net462\FundApps.Common.dll
	  FundApps.Rapptr.Web -> C:\Development\Sample\src\FundApps.Rapptr.Web\bin\FundApps.Rapptr.Web.dll

	Build succeeded.
	    0 Warning(s)
	    0 Error(s)

	Time Elapsed 00:00:11.91

## Environment data
`dotnet --info` output:

.NET Command Line Tools (2.0.2)

Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.2\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

**Comment by livarcocc** [comment]
It seems to me like you are trying to bring in your own compilers, is that really the case here?

**Comment by AdamPew** [comment]
Have you managed to solve the issue on the first build error?

**Comment by livarcocc** [comment]
I have not, but I am trying to understand what is it that is being tried here.

**Comment by jamescrowley** [comment]
@livarcocc nope - just trying to make sure C# 7 features are supported in Razor views :)

**Comment by jamescrowley** [comment]
@AdamPew no, I got blocked by this and was intending to come back and try again at a later stage.

**Comment by livarcocc** [comment]
Closing due to lack of activity.

**Comment by jayparmar** [comment]
This is happening again. Can we please have a proper fix and not just close the issue for shake of closing it. Thanks

**Comment by livarcocc** [comment]
If you have a consistent repro for this, please share with us repro steps and perhaps a repro repo on github that we can clone and use to investigate this issue further. If that's the case, I suggest you file a new issue for it with the information I requested above.

As for closing issues that have not have activity on it or where we asked questions that we did not get answers for in a good time, that's something that we will continue to do in the repo to prevent stale issues from lingering and making it difficult to understand what we are working on and still considering. There are already enough of those kinds of issues around.
